### PR TITLE
docs(authentication): list official authentication strategies, clarify app authentication example

### PR DESCRIPTION
**Pull request by gr2m**
addresses #1893 /cc @webbertakken

-----
[View rendered docs/src/pages/api/01_authentication.md](https://github.com/octokit/rest.js/blob/1893/docs-clarify-authentication/docs/src/pages/api/01_authentication.md)

### CODE REVIEW

File: docs/src/pages/api/01_authentication.md [hunk 1]
```diff
@@ -4,7 +4,16 @@ title: "Authentication"
...
 
-By default, Octokit authenticates using the [token authentication strategy](https://github.com/octokit/auth-token.js). Pass in a token using `options.auth`. It can be a personal access token, an OAuth token, an installation access token or a JSON Web Token for GitHub App authentication. The `Authorization` header will be set according to the type of token.
+GitHub supports different authentication strategies:
+
+1. **Personal access token** ([create](https://github.com/settings/tokens/new)). This is the default authentication strategy. Set the `options.auth` option to the token in `new Octokit(options)`. Learn more about the built-in [`@octokit/auth-token` authentication strategy](https://github.com/octokit/auth-token.js).
+2. **OAuth Apps**: authenticate using user access token created by an OAuth app, to which you granted selected permissions, or as the OAuth App itsel (OAuth using `client_id` and `client_secret`). Learn more about the optional [`@octokit/auth-oauth-app` authentication strategy](https://github.com/octokit/auth-oauth-app.js)
```
Comment (wolfy1339) [inline docs/src/pages/api/01_authentication.md]: Small typo
```suggestion
2. **OAuth Apps**: authenticate using user access token created by an OAuth app, to which you granted selected permissions, or as the OAuth App itself (OAuth using `client_id` and `client_secret`). Learn more about the optional [`@octokit/auth-oauth-app` authentication strategy](https://github.com/octokit/auth-oauth-app.js)
```

Comment (gr2m) [reply docs/src/pages/api/01_authentication.md]: thanks!

File: docs/src/pages/api/01_authentication.md [hunk 2]
```diff
@@ -29,6 +38,9 @@ const appOctokit = new Octokit({
   auth: {
     id: 123,
     privateKey: process.env.PRIVATE_KEY,
+    // optional: this will make appOctokit authenticate as app (JWT)
+    //           or installation (access token), depending on the request URL
+    installationId: 123,
```
Comment (webbertakken) [inline docs/src/pages/api/01_authentication.md:43]: yes, this part would definitely have helped for sure!

Comment (gr2m) [reply docs/src/pages/api/01_authentication.md:43]: thanks for the review 👍🏼

**Review by oscard0m — COMMENTED** [review]
❤️

**Pull request merged**